### DISCUSSION TITLE
detectHostRulesFromEnv ony supports domain names, should also support URLs with paths

### DISCUSSION THREAD

**Discussion by reegnz** [renovatebot/renovate#32121] [Suggest an Idea]
### Tell us more.

I want to use self-hosted renovate with `detectHostRulesFromEnv`. 
I want to detect private docker images from `ghrc.io/myprivaterepo`. Currently it seems to me like I can only set `DOCKER_GHCR_IO_TOKEN=the-token` which is for the entirety of `ghcr.io`, not scoped to just my org.

Is there a way to pass a token to a URL name like `ghcr.io/myprivaterepo` through `detectHostRulesFromEnv`?
I don't want to use the token for all `ghcr.io` access, just for particular repos.

I'm aware that I can pass this with a host-rule in the json config, I'm just wondering if it would be doable with env vars only.

**Comment by rarkins** [comment]
It's not supported today. You would need to figure out a way to support this syntax in env variables, because `/` is not supported directly

**Reply by reegnz** [reply]
Yeah, that's what I've figured, it's pretty difficult to represent it as an env var. Maybe a combination of env-vars though, one declaring an alias, and then use alias in the other env var, eg:

```
HOST_ALIAS_PRIVATEGHCR="ghcr.io/myprivaterepo"
DOCKER_PRIVATEGHCR_TOKEN=mytoken
```

Then the logic could first load the aliases, then try to resolve the alias before evaluating the second env var.

**Reply by rarkins** [reply]
I think you'd be better off to just stringing your hostRules and expose them as RENOVATE_HOST_RULES

**Reply by reegnz** [reply]
Yeah, probably. Thanks for that hint.
